BackendApartmentReservation/BackendApartmentReservation/Controllers/ProfilesController.cs

```diff
@@ -8,11 +16,38 @@ namespace BackendApartmentReservation.Controllers
     [ApiController]
     public class ProfilesController : ControllerBase
     {
+        private readonly UserManager<DbEmployee> _userManager;
         [HttpGet]
         [Route("profiles")]
         public async Task<IEnumerable<string>> Get()
         {
             return await Task.FromResult(new[] { "userProfile1", "userProfile2" });
         }
+
+        [HttpPost]
+        [AllowAnonymous]
+        [Route("register")]
+        public async Task<IActionResult> Register([FromBody] RegisterRequest model)
+        {
+            if (!ModelState.IsValid) return BadRequest(ModelState);
```

Comment:
Regarding model state validation. It is very tedious to manually validate it before every request. Use validation attributes (as you are using `Required` for example) and let request pipline do validation and handle `BadRequest` response for you.

Take a look at my implementation of `IAsyncActionFilter` (implemented by `MethodCallLoggingFilter`). Would be nice if you wrote something similar which is called before invocation and automatically rejects request before execution.

Reply:
Understand the problem, but not sure how the implementation should look. `` MethodCallLoggingFilter `` seems to work similarly to  what it is here.  Also, can't find annotations for request validations. All the online examples seem to favor `` if (!ModelState.IsValid) return BadRequest(ModelState); `` .

Reply:
Yes, you wont avoid using `if (!ModelState.IsValid)` :) but instead of writing it in every controller, you can write one action filter that will do that before every request.

Reply:
So what you could do: write something like `ValidationFilter` which validates model state, and logs validation errors if its not valid. Also, it will already set a response, so action will not be executed if model is not valid